subsys/bluetooth/audio/audio_iso.h

```diff
@@ -0,0 +1,62 @@
+/* @file
+ * @brief Internal APIs for Audio ISO
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/iso.h>
+#include <zephyr/bluetooth/audio/audio.h>
+
+/** Audio ISO Type */
+enum bt_audio_iso_type {
+	BT_AUDIO_ISO_TYPE_UNICAST = BIT(0),
+	BT_AUDIO_ISO_TYPE_BROADCAST = BIT(1),
+	BT_AUDIO_ISO_TYPE = BT_AUDIO_ISO_TYPE_UNICAST |
+			    BT_AUDIO_ISO_TYPE_BROADCAST,
+} __packed;
+
+struct bt_audio_iso {
+	enum bt_audio_iso_type type;
+	struct bt_iso_chan iso_chan;
+	struct bt_iso_chan_qos iso_qos;
+	struct bt_iso_chan_io_qos sink_io_qos;
+	struct bt_iso_chan_io_qos source_io_qos;
+	struct bt_iso_chan_path sink_path;
+	/* TODO: The sink/source path CC will basically be a duplicate of
+	 * bt_codec.data, but since struct bt_codec stores the information as an
+	 * array of bt_codec_data, and ISO expect a uint8_t array, we need to
+	 * duplicate the data. This should be optimized.
+	 */
+	uint8_t sink_path_cc[CONFIG_BT_CODEC_MAX_DATA_COUNT * CONFIG_BT_CODEC_MAX_DATA_LEN];
+	struct bt_iso_chan_path	source_path;
+	uint8_t source_path_cc[CONFIG_BT_CODEC_MAX_DATA_COUNT * CONFIG_BT_CODEC_MAX_DATA_LEN];
+
+	union {
+		struct {
+			struct bt_audio_stream *stream;
+		} broadcast;
+		struct {
+			struct bt_audio_stream *sink_stream;
+			struct bt_audio_stream *source_stream;
+		} unicast;
```

Comment:
I found some issues having struct bt_audio_stream * instead of struct bt_audio_ep * inside of bt_audio_iso.
I'm about to change it, to store EP pointer here